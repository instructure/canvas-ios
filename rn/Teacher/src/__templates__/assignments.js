//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: '1376',
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
  anonymize_students: false,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment({ assignment_group_id: '1' })],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
