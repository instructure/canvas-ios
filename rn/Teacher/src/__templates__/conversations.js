//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const conversation: Template<Conversation> = template({
  id: '123456789',
  subject: 'learn to write code',
  workflow_state: 'read',
  last_message: 'this is the worst class ever',
  last_message_at: '2037-06-01T05:59:00Z',
  last_authored_message: 'this is the worst class ever',
  last_authored_message_at: '2037-06-01T05:59:00Z',
  participants: [
    {
      id: '1234',
      name: 'artorias of the abyss',
    },
    {
      id: '6789',
      name: 'solaire of astora',
    },
  ],
  avatar_url: 'http://www.fillmurray.com/200/300',
  context_name: 'Test Class',
  context_code: 'course_1',
  audience_contexts: {},
  audience: [],
  message_count: 2,
  private: false,
  properties: null,
  starred: false,
  subscribed: false,
  visible: true,
})

export const conversationMessage: Template<ConversationMessage> = template({
  id: '123456789',
  body: 'this is neat',
  author_id: '1234',
  created_at: '2037-06-01T05:59:00Z',
  attachments: [
    {
      id: '1',
      display_name: 'something.pdf',
      mime_class: 'pdf',
    },
  ],
  generated: false,
})
