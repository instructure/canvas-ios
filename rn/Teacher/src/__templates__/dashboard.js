//
// This file is part of Canvas.
// Copyright (C) 2019-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const dashboardCardLink: Template<DashboardCardLink> = template({
  css_class: 'assignments',
  hidden: null,
  icon: 'icon-assignment',
  label: 'Assignments',
  path: '/courses/1/assignments',
})

export const dashboardCard: Template<DashboardCard> = template({
  assetString: 'course_1',
  courseCode: 'MET-132',
  enrollmentType: 'StudentEnrollment',
  href: '/courses/1',
  id: '1',
  image: null,
  links: [dashboardCardLink()],
  longName: 'A good course - MET',
  originalName: 'A good course',
  position: null,
  shortName: 'A good course',
  subtitle: 'enrolled as: Student',
  term: null,
})
