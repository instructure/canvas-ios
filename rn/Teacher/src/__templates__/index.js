//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
