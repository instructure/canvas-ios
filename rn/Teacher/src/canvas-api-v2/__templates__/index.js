// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
