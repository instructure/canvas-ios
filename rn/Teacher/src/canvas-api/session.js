//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

let currentSession: ?Session = null

export function setSession (session: ?Session): void {
  currentSession = session
}

export function getSession (): Session {
  if (!currentSession) {
    console.warn('The user session is being requested, yet none exists. Are you sure that is what you want?')
    // The main usecase for actually returning this blank session is when components render on logout
    // On logout, we clear out redux which triggers components to render, which in turn those components might need a session
    // But, blank session is fine because the user has logged out
    return {
      baseURL: '',
      authToken: '',
      user: {
        id: '',
        primary_email: '',
        avatar_url: '',
        name: '',
      },
    }
  }
  return currentSession
}

// There are a few places where we do need to get the session and know whether it exists, but that's rare
// Almost everywhere should use getSession() instead
export function getSessionUnsafe (): ?Session {
  return currentSession
}

// returns true is the sessions are similar, otherwise returns false
export function compareSessions (s1: Session, s2: Session): boolean {
  return (
    s1.baseURL === s2.baseURL &&
    s1.user.id === s2.user.id &&
    s1.actAsUserID === s2.actAsUserID
  )
}
