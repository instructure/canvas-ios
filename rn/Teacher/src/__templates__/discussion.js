//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'
import { assignment } from './assignments'
import { userDisplay } from './users'

export const discussion: Template<Discussion> = template({
  id: '1',
  assignment_id: assignment().id,
  title: 'Dear Ivy Iversen, your last name cannot start with your first name',
  html_url: 'http://mobiledev.instructure.com/courses/1/discussion_topics',
  pinned: false,
  position: 1,
  posted_at: '2016-11-11T04:03:17Z',
  published: true,
  read_state: 'read',
  sort_by_rating: false,
  subscribed: true,
  user_can_see_posts: true,
  user_name: 'Ivy Iversen',
  unread_count: 1,
  discussion_subentry_count: 2,
  permissions: {
    'attach': true,
    'delete': true,
    'reply': true,
    'update': true,
  },
  message: '<p>And why?</p>',
  assignment: assignment(),
  last_reply_at: '2016-12-11T04:03:17Z',
  author: userDisplay(),
  allow_rating: false,
  only_graders_can_rate: false,
  require_initial_post: false,
  delayed_post_at: null,
  discussion_type: 'side_comment',
  unlock_at: null,
  can_unpublish: true,
  is_section_specific: false,
})

export const discussionReply: Template<DiscussionReply> = template({
  'id': '25472827',
  'user_id': userDisplay().id,
  'parent_id': null,
  'created_at': '2017-05-23T17:12:04Z',
  'updated_at': '2017-05-23T17:12:04Z',
  'rating_count': null,
  'rating_sum': null,
  'message': '<p>1.0</p>',
  'replies': [],
})

export const discussionEditReply: Template<DiscussionReply> = template({
  'id': '25472827',
  'editor_id': '1',
  'user_id': '1',
  'parent_id': null,
  'created_at': '2017-05-23T17:12:04Z',
  'updated_at': '2017-05-23T17:12:04Z',
  'rating_count': null,
  'rating_sum': null,
  'message': '<p>2.0</p>',
  'replies': [],
})

export const discussionView: Template<DiscussionView> = template({
  'new_entries': [],
  'unread_entries': [
    '25458826',
    '25458830',
  ],
  entry_ratings: {},
  participants: [userDisplay()],
  'view': [
    discussionReply({
      'id': '25472827',
      'user_id': '1',
      'parent_id': null,
      'created_at': '2017-05-23T17:12:04Z',
      'updated_at': '2017-05-23T17:12:04Z',
      'rating_count': null,
      'rating_sum': null,
      'message': '<p>1.0</p>',
      'replies': [],
    }),
  ],
})

export const discussionViewLarge: Template<DiscussionView> = template({
  'new_entries': [],
  'unread_entries': [
    '25458826',
    '25458830',
  ],
  'entry_ratings': {
    '4': 1,
  },
  participants: [userDisplay()],
  'view': [
    discussionReply({
      'id': '25472827',
      'user_id': '5347622',
      'parent_id': null,
      'created_at': '2017-05-23T17:12:04Z',
      'updated_at': '2017-05-23T17:12:04Z',
      'rating_count': null,
      'rating_sum': null,
      'message': '<p>1.0</p>',
      'replies': [
        discussionReply({
          'id': '25472860',
          'user_id': '5347622',
          'parent_id': '25472827',
          'created_at': '2017-05-23T17:14:43Z',
          'updated_at': '2017-05-23T17:14:43Z',
          'rating_count': null,
          'rating_sum': null,
          'message': '<p>1.1</p>',
        }),
        discussionReply({
          'id': '25472862',
          'user_id': '5347622',
          'parent_id': '25472827',
          'created_at': '2017-05-23T17:14:51Z',
          'updated_at': '2017-05-23T17:14:51Z',
          'rating_count': null,
          'rating_sum': null,
          'message': '<p>1.2</p>',
        }),
      ],
    }),
    discussionReply({
      'id': '25472831',
      'user_id': '5347622',
      'parent_id': null,
      'created_at': '2017-05-23T17:12:15Z',
      'updated_at': '2017-05-23T17:12:15Z',
      'rating_count': null,
      'rating_sum': null,
      'message': '<p>2.0</p>',
      'replies': [
        {
          'id': '25472941',
          'user_id': '5347622',
          'parent_id': '25472831',
          'created_at': '2017-05-23T17:20:43Z',
          'updated_at': '2017-05-23T17:20:43Z',
          'rating_count': null,
          'rating_sum': null,
          'message': '<p>Myself</p>',
          'attachment': {
            'id': '107777279',
            'folder_id': '6852597',
            'display_name': 'profile_photo.jpg',
            'filename': 'profile_photo.jpg',
            'content-type': 'image/jpeg',
            'url': 'https://mobiledev.instructure.com/files/107777279/download?download_frd=1&verifier=IL9TUHoM8P5RsPUSASyZEYv6mOkhAqUR3fBOPvt5',
            'size': 416708,
            'created_at': '2017-05-23T17:20:43Z',
            'updated_at': '2017-05-23T17:20:43Z',
            'unlock_at': null,
            'locked': false,
            'hidden': false,
            'lock_at': null,
            'hidden_for_user': false,
            'thumbnail_url': 'https://instructure-uploads.s3.amazonaws.com/account_99298/thumbnails/88538243/profile_photo_thumb.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJFNFXH2V2O7RPCAA%2F20170523%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20170523T172053Z&X-Amz-Expires=518400&X-Amz-SignedHeaders=host&X-Amz-Signature=d67a6925ea4316318cd5328baa8db3199b41cd3f5119acbda3e0feddf7bfb225',
            'modified_at': '2017-05-23T17:20:43Z',
            'mime_class': 'image',
            'media_entry_id': null,
            'locked_for_user': false,
          },
        },
      ],
    })],
})

export const createDiscussionParams: Template<CreateDiscussionParameters> = template({
  title: 'Infernal Shrines',
  message: 'Summon a mighty Punisher',
  allow_rating: false,
  only_graders_can_rate: false,
  sort_by_rating: false,
  require_initial_post: false,
  delayed_post_at: null,
  is_announcement: false,
})

export const updateDiscussionParams: Template<UpdateDiscussionParameters> = template({
  id: '1',
  ...createDiscussionParams(),
})

export const discussionPermissions: Template<CoursePermissions> = template({
  create_announcement: true,
  create_discussion_topic: true,
})
