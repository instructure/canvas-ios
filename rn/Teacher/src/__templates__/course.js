//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import { CourseModel } from '../canvas-api/model'
import template, { type Template } from '../utils/template'
import { section } from './section'

export const courseTerm: Template<CourseTerm> = template({
  id: '1',
  created_at: '2017-03-17T19:15:25Z',
  name: 'Default Term',
})

export const course: Template<Course> = template({
  id: '1',
  account_id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
  permissions: {
    viewAnalytics: true,
    viewAllGrades: true,
  },
  workflow_state: 'available',
  term: {
    id: '1',
    created_at: '2018-02-08T23:13:39Z',
    end_at: '2030-02-08T23:13:39Z',
    start_at: '2018-02-08T23:13:39Z',
    name: 'Default Term',
    workflow_state: 'available',
  },
  end_at: '2030-02-08T23:13:39Z',
})

export const courseModel: Template<CourseModel> = overrides =>
  Object.assign(new CourseModel(course()), overrides)

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})

export const courseSettings: Template<CourseSettings> = template({
  allow_student_forum_attachments: true,
})
