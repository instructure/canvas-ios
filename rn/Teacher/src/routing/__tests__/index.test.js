//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import { shallow } from 'enzyme'
import React from 'react'
import { View, Text } from 'react-native'
import { createStore } from 'redux'
import { wrapComponentInProviders, route } from '../'
import { setSession } from '../../canvas-api/session.js'
import * as template from '../../__templates__'

describe('wrapComponentInProviders', () => {
  const store = createStore(() => ({}))

  it('renders wrapped screen with store correctly', () => {
    const session = template.session({ actAsUserID: 2 })
    setSession(session)
    const generator = () => () =>
      <View>
        <Text>Test Screen</Text>
      </View>
    const wrappedGenerator = wrapComponentInProviders('TestScreen', generator, store)
    const Wrapped = wrappedGenerator()

    let tree = shallow(<Wrapped />)
    expect(tree).toMatchSnapshot()
  })

  it('renders wrapped screen and shows the error screen when there is an error', () => {
    global.crashReporter = { recordError: jest.fn(), setString: jest.fn(), setBool: jest.fn() }
    const ErrorThrower = () => { throw new Error() }
    const generator = () => ErrorThrower
    const wrappedGenerator = wrapComponentInProviders('TestScreen', generator, store)
    const Wrapped = wrappedGenerator()

    let tree = shallow(<Wrapped />)
    try { // https://github.com/airbnb/enzyme/issues/1255
      tree.find(ErrorThrower).dive()
    } catch (err) {
      tree.instance().componentDidCatch(err)
      tree.update()
    }
    expect(global.crashReporter.recordError).toHaveBeenCalled()
    expect(tree).toMatchSnapshot()
  })
})

test('route to something that does not exist', () => {
  expect(route('garbage')).toBeUndefined()
})
