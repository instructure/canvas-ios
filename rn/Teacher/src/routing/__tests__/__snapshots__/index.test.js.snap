// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapComponentInProviders renders wrapped screen and shows the error screen when there is an error 1`] = `
<ErrorScreen
  navigator={
    Navigator {
      "isModal": undefined,
      "moduleName": "TestScreen",
    }
  }
/>
`;

exports[`wrapComponentInProviders renders wrapped screen with store correctly 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": HttpLink {
        "request": [Function],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": HttpLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.3",
      "watchQuery": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component
      apollo={
        ApolloClient {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": HttpLink {
            "request": [Function],
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "queryManager": QueryManager {
            "assumeImmutableResults": false,
            "clientAwareness": Object {
              "name": undefined,
              "version": undefined,
            },
            "dataStore": DataStore {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "fetchQueryRejectFns": Map {},
            "idCounter": 1,
            "inFlightLinkObservables": Map {},
            "link": HttpLink {
              "request": [Function],
            },
            "localState": LocalState {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "client": [Circular],
            },
            "mutationStore": MutationStore {
              "store": Object {},
            },
            "onBroadcast": [Function],
            "pollingInfoByQueryId": Map {},
            "queries": Map {},
            "queryDeduplication": true,
            "queryStore": QueryStore {
              "store": Object {},
            },
            "ssrMode": false,
            "transformCache": WeakMap {},
          },
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.6.3",
          "watchQuery": [Function],
        }
      }
      navigator={
        Navigator {
          "isModal": undefined,
          "moduleName": "TestScreen",
        }
      }
      session={
        Object {
          "actAsUserID": 2,
          "authToken": "iamanauthtoken",
          "baseURL": "http://mobiledev.instructure.com/",
          "user": Object {
            "avatar_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
            "id": "1",
            "name": "Key and Peele",
            "primary_email": "keyandpeele@instructure.com",
          },
        }
      }
      testID="TestScreen"
    />
  </Provider>
</ApolloProvider>
`;
